src/map/entities/baseentity.cpp

```diff
@@ -183,12 +183,27 @@ void CBaseEntity::ResetLocalVars()
 
 uint32 CBaseEntity::GetLocalVar(const char* var)
 {
-    return m_localVars[var];
+    if (sizeof(var) && m_localVars[var])
```

Comment:
`sizeof` as a safety measure for a `nullptr` being passed in here won't work. `sizeof` returns the size of the type of the object that its being given, in this case its a pointer, so it'll always be 8 on a 64-bit platform, even when its `nullptr`.

The trouble here is when a `nullptr` `const char*` is then passed into `m_localVars[var]`, since this is `std::map<std::string, std::uint32_t> m_localVars;`, it'll try to construct `var` into a `std::string`, where it'll explode because it's presumably `nullptr`.

Reply:
Also, map and unordered_map will CREATE an entry when you use operator[], so if you do `if (m_localVars[var]) ...` you're creating an entry and then returning it, so it'll always be true

Reply:
Removed this check and set it back to what it was originally, along with the check below on setlocalvar as well.